src/http/request.cr

```diff
@@ -148,21 +147,26 @@ class HTTP::Request
       end
     end
 
-    request_line = io.gets(4096, chomp: true)
+    request_line = io.gets(HTTP.max_request_line_size + 1, chomp: true)
     return nil unless request_line
 
+    # Identify Request-URI too long
+    if request_line.bytesize > HTTP.max_request_line_size
```

Comment:
I'd assign `HTTP.max_request_line_size` to a variable, in order to save couple of calls here.

```suggestion
    if request_line.bytesize > HTTP.max_request_line_size
```